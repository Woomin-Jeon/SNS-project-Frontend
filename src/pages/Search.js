import React, {useEffect} from 'react';
import { Link } from 'react-router-dom';
import HeaderOthersPage from '../components/headers/HeaderOthersPage';
import SearchBox from '../components/headers/SearchBox';
import { checkSessionExist } from '../function';

const callAPI = async (currentUserState, setCurrentUserState) => {
  const { user } = await checkSessionExist();

  setCurrentUserState({
    ...currentUserState,
    id: user.id,
    userName: user.userName,
    friends: user.friends,
    profile: user.profile,
  });
};

function Search({
  searchState,
  setSearchState,
  topLevelState,
  setTopLevelState,
  loginState,
  setLoginState,
  currentUserState,
  setCurrentUserState,
}) {
  const { users } = loginState;

  useEffect(() => {
    callAPI(currentUserState, setCurrentUserState);
  }, []);

  const findUserById = (id) => {
    for (let i = 0; i < users.length; i += 1) {
      if (id === users[i].id) {
        setTopLevelState({
          ...topLevelState,
          id: users[i].id,
          name: users[i].userName,
        });
      }
    }
    setSearchState({ ...setSearchState, contents: [], exist: false });
  };

  return (
    <>
      <div className="header">
        <SearchBox
          loginState={loginState}
          setLoginState={setLoginState}
          setSearchState={setSearchState}
        />
        <HeaderOthersPage
          loginState={loginState}
          setLoginState={setLoginState}
          currentUserState={currentUserState}
          setCurrentUserState={setCurrentUserState}
        />
      </div>
      <br />
      <br />
      <div>
        <div className="join-following">
          {searchState.exist
            ? (
              searchState.contents.map((v, index) => (
                <div key={index}>
                  <label className="join-users">
                    <img className="join-user-profile" src={v.profile} alt="" />
                    <div className="join-users-informatrion">
                      <Link
                        to="/otherspage"
                        className="join-users-name"
                        type="button"
                        onClick={() => findUserById(v.id)}
                      >
                        {v.userName}
                      </Link>
                      <div className="join-users-lint">{v.birth}</div>
                      <div className="join-users-lint">{v.location}</div>
                      <div className="join-users-lint">{v.email}</div>
                    </div>
                  </label>
                </div>)))
            : (<span>해당 유저가 존재하지 않습니다</span>)}
        </div>
      </div>
    </>
  );
}

export default Search;
