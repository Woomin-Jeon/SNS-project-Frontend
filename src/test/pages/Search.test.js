import Search from "../../pages/Search";
import React from "react";
import {shallow} from 'enzyme';

describe('Search', () => {
  let currentUserState;
  let setCurrentUserState;
  let loginState;
  let setLoginState;
  let postState;
  let topLevelState;
  let setTopLevelState;
  let searchState;
  let setSearchState;

  beforeEach(() => {
    postState = { post: [] };
    loginState = { users: [] };
  });

  it('renders well', () => {
    const component = shallow(
      <Search
        searchState={searchState}
        setSearchState={setSearchState}
        topLevelState={topLevelState}
        setTopLevelState={setTopLevelState}
        loginState={loginState}
        setLoginState={setLoginState}
        currentUserState={currentUserState}
        setCurrentUserState={setCurrentUserState}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
