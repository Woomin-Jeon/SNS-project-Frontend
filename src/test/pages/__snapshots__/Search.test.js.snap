// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders well 1`] = `ShallowWrapper {}`;
